go/vt/key/key.go

```diff
@@ -49,261 +52,258 @@ func (i Uint64Key) Bytes() []byte {
 	return buf
 }
 
-//
-// KeyRange helper methods
-//
+// Helper methods for keyspace id values.
 
-// EvenShardsKeyRange returns a key range definition for a shard at index "i",
-// assuming range based sharding with "n" equal-width shards in total.
-// i starts at 0.
-//
-// Example: (1, 2) returns the second out of two shards in total i.e. "80-".
-//
-// This function must not be used in the Vitess code base because Vitess also
-// supports shards with different widths. In that case, the output of this
-// function would be wrong.
-//
-// Note: start and end values have trailing zero bytes omitted.
-// For example, "80-" has only the first byte (0x80) set.
-// We do this to produce the same KeyRange objects as ParseKeyRangeParts() does.
-// Because it's using the Go hex methods, it's omitting trailing zero bytes as
-// well.
-func EvenShardsKeyRange(i, n int) (*topodatapb.KeyRange, error) {
-	if n <= 0 {
-		return nil, fmt.Errorf("the shard count must be > 0: %v", n)
-	}
-	if i >= n {
-		return nil, fmt.Errorf("the index of the shard must be less than the total number of shards: %v < %v", i, n)
-	}
-	if n&(n-1) != 0 {
-		return nil, fmt.Errorf("the shard count must be a power of two: %v", n)
+// Normalize adds padding to id as necessary to make sure it can be compared as an 8 byte integer.
+// From a key range perspective, -80 == 00-80 == 0000-8000 == 000000-800000, etc. and they should
+// always be treated the same even if they are different lengths.
```

Comment:
I also thought about it like `-ff` but those are just weird cases, Definitely, they can be handled though.

Reply:
Due to the recent changes to remove trailing zeroes rather than add padding, the `0` case and `00-` case are handled automatically (munged into the same thing as `-`). I don't think there's anything special about `-ff` at all (it's just a regular value).